pymc/tests/test_distributions_timeseries.py

```diff
@@ -12,20 +12,155 @@
+
+        combined_kwargs = {**default_kwargs, **kwargs}
+        grw = gaussianrandomwalk(
+            combined_kwargs["mu"],
+            combined_kwargs["sd"],
+            combined_kwargs["init"],
+            combined_kwargs["steps"],
+        ).eval()
+
+        assert grw.shape == expected
+
+    def test_grw_logp(self):
+        # `at.diff` is currently broken with constants
+        test_vals = [0, 1, 2]
+        vals = at.vector("vals")
+        mu = 1
+        sigma = 2
+        init = pm.Normal.dist(0, sigma)
+
+        with pm.Model():
+            grw = GaussianRandomWalk("grw", mu, sigma, init=init, steps=2)
+
+        logp = pm.logp(grw, vals)
+        logp_eval = logp.eval({vals: test_vals})
+
+        logp_reference = (
+            stats.norm(0, sigma).logpdf(test_vals[0])
+            + stats.norm(mu, sigma).logpdf(np.diff(test_vals)).sum()
+        )
+
+        np.testing.assert_almost_equal(logp_eval, logp_reference, decimal=6)
+
+    def test_grw_inference(self):
+        mu, sigma, steps = 2, 1, 10000
+        obs = np.concatenate([[0], np.random.normal(mu, sigma, size=steps)]).cumsum()
+
+        with pm.Model():
+            _mu = pm.Uniform("mu", -10, 10)
+            _sigma = pm.Uniform("sigma", 0, 10)
+            # Workaround for bug in `at.diff` when data is constant
+            obs_data = pm.MutableData("obs_data", obs)
+            grw = GaussianRandomWalk("grw", _mu, _sigma, steps=steps, observed=obs_data)
+
+            trace = pm.sample()
+
+        recovered_mu = trace.posterior["mu"].mean()
+        recovered_sigma = trace.posterior["sigma"].mean()
+        np.testing.assert_allclose([mu, sigma], [recovered_mu, recovered_sigma], atol=0.2)
+
+    @pytest.mark.parametrize(
+        "steps,size,expected",
+        (
+            (1, None, (2,)),
+            (2, 1, (1, 3)),
+            (2, 5, (5, 3)),
+            (10, 5, (5, 11)),
+        ),
+    )
+    def test_grw_shape(self, steps, size, expected):
+        grw_dist = pm.GaussianRandomWalk.dist(mu=0, sigma=1, steps=steps, size=size)
+        expected_symbolic = tuple(grw_dist.shape.eval())
+        assert expected_symbolic == expected
+
+    @pytest.mark.parametrize("size", (None, (1, 2), (10, 2), (3, 100, 2)))
+    def test_init_automatically_resized(self, size):
+        x = GaussianRandomWalk.dist(mu=[0, 1], init=pm.Normal.dist(), size=size)
+        init = x.owner.inputs[-2]
+        assert init.eval().shape == size if size is not None else (2,)
+
+        x = GaussianRandomWalk.dist(mu=[0, 1], init=pm.Normal.dist(size=5), shape=size)
+        init = x.owner.inputs[-2]
+        assert init.eval().shape == size if size is not None else (2,)
+
+
+class TestGaussianRandomWalk(BaseTestDistributionRandom):
+    # Override default size for test class
+    size = None
+
+    pymc_dist = pm.GaussianRandomWalk
+    pymc_dist_params = {"mu": 1.0, "sigma": 2, "init": pm.Constant.dist(0), "steps": 4}
```

Comment:
@ricardoV94 regarding the earlier comment its because we did this. Init doesn't support int arguments anymore given the other changes we added

Reply:
What happens if you omit init?

Reply:
See https://github.com/pymc-devs/pymc/pull/5298#issuecomment-1065356038